src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -916,7 +930,7 @@ private static JsonRpcMessage CreateError(JToken id, Exception exception)
 
                 return await ((Task)result).ContinueWith(this.handleInvocationTaskResultDelegate, request.Id, TaskScheduler.Default).ConfigureAwait(false);
             }
-            catch (Exception ex)
+            catch (Exception ex) when (!this.IsFatalException(ex.InnerException ?? ex))
```

Comment:
This redirection to the InnerException if it's defined makes me nervous that we might disregard an outer exception that is useful. I expect you're trying to strip off the `TargetInvocationException` and/or `AggregateException`, is that right? In that case, can we add new catch blocks above this one for those specific types and strip the outer exception just for those, leaving any other exceptions alone?

Reply:
Or perhaps since you want to strip it in other places as well that don't use catch blocks, you can write a helper method that does the type checking on the exception and strips it only when it's appropriate.